src/adapter/src/catalog/storage.rs

```diff
@@ -385,6 +412,151 @@ async fn migrate(
+        //
+        // TODO(jkosh44) Can be cleared (patched to be empty) in v0.50.0
+        |txn: &mut Transaction<'_>, now, _bootstrap_args| {
+            // Delete all system roles so we can re-insert them with known IDs.
+            let sys_roles = txn.roles.delete(|role_key, _| role_key.id.is_system());
+            assert_eq!(2, sys_roles.len(), "unexpected number of system roles");
+            txn.roles.insert(
+                RoleKey {
+                    id: MZ_SYSTEM_ROLE_ID,
+                },
+                RoleValue {
+                    role: (&*MZ_SYSTEM_ROLE).into(),
+                },
+            )?;
+            txn.roles.insert(
+                RoleKey {
+                    id: MZ_INTROSPECTION_ROLE_ID,
+                },
+                RoleValue {
+                    role: (&*MZ_INTROSPECTION_ROLE).into(),
+                },
+            )?;
+
+            // We need a default owner for all existing objects. So we just create a new role for
+            // this purpose named "default_owner". If one happens to exist with the same name, then
+            // just use that.
+            let default_owner_name = "default_owner";
+            let default_owner_id = match txn
+                .roles
+                .items()
+                .iter()
+                .filter(|(_, role_value)| role_value.role.name == default_owner_name)
+                .next()
+            {
+                Some((role_key, _)) => role_key.id,
+                None => {
+                    let role_id = txn.insert_user_role(SerializedRole {
+                        name: default_owner_name.to_string(),
+                        attributes: RoleAttributes::new(),
+                        membership: Some(RoleMembership::new()),
+                    })?;
+                    let audit_id = txn.get_and_increment_id(AUDIT_LOG_ID_ALLOC_KEY.to_string())?;
+                    txn.audit_log_updates.push((
+                        AuditLogKey {
+                            event: VersionedEvent::new(
+                                audit_id,
+                                EventType::Create,
+                                ObjectType::Role,
+                                EventDetails::IdNameV1(mz_audit_log::IdNameV1 {
+                                    id: role_id.to_string(),
+                                    name: default_owner_name.to_string(),
+                                }),
+                                None,
+                                now,
+                            ),
+                        },
+                        (),
+                        1,
+                    ));
+                    role_id
+                }
+            };
+
+            txn.databases.update(|database_key, database_value| {
+                let mut database_value = database_value.clone();
+                if database_value.owner_id.is_none() {
+                    if database_key.id == MATERIALIZE_DATABASE_ID {
```

Comment:
This is a bit surprising. Why is the materialize db special? Is it because it must always exist, but there's no users yet, so we have to assign it some owner, and that's the system?

Reply:
> Is it because it must always exist, but there's no users yet, so we have to assign it some owner, and that's the system?

That's the exact reason.